src/iarc7_motion/iarc_tasks/velocity_test_task.py

```diff
@@ -0,0 +1,128 @@
+            rospy.logerr('Could not lookup a parameter for constant velocity test task')
+            raise
+
+        self._z_holder = HeightHolder(True)
+        self._height_checker = HeightSettingsChecker()
+        self._limiter = AccerlationLimiter()
+
+        self._state = VelocityTestTaskState.init
+
+    def _current_velocity_callback(self, data):
+        with self._lock:
+            self._drone_odometry = data
+
+    def get_desired_command(self):
+        with self._lock:
+            if (self._state == VelocityTestTaskState.init):
+                if self._drone_odometry is None:
+                    self._state = VelocityTestTaskState.waiting
+                else:
+                    self._state = VelocityTestTaskState.moving
+                return (TaskRunning(), NopCommand())
+
+            elif (self._state == VelocityTestTaskState.waiting):
+                if self._drone_odometry is None:
+                    self._state = VelocityTestTaskState.waiting
+                else:
+                    self._state = VelocityTestTaskState.moving
+                return (TaskRunning(), NopCommand())
+
+            elif self._canceled:
+                return (TaskCanceled(),)
+
+            elif self._state == VelocityTestTaskState.moving:
+
+                if not (self._height_checker.above_min_maneuver_height(
+                            self._drone_odometry.pose.pose.position.z)):
+                    return (TaskAborted(msg='Drone is too low'),)
+
+                x_vel_target = self._TRANSLATION_SPEED_X
+                y_vel_target = self._TRANSLATION_SPEED_Y
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+                desired_vel = []
+                desired_vel.append(x_vel_target)
+                desired_vel.append(y_vel_target)
+                desired_vel.append(z_vel_target)
+
+                curr_vel = []
+                curr_vel.append(self._drone_odometry.twist.twist.linear.x)
+                curr_vel.append(self._drone_odometry.twist.twist.linear.y)
+                curr_vel.append(self._drone_odometry.twist.twist.linear.z)
+
+                desired_vel = self._limiter.limit_acceleration(curr_vel, desired_vel)
+
+                velocity = TwistStamped()
+                velocity.header.frame_id = 'level_quad'
+                velocity.header.stamp = rospy.Time.now()
+                velocity.twist.linear.x = desired_vel[0]
+                velocity.twist.linear.y = desired_vel[1]
+                velocity.twist.linear.z = desired_vel[2]
+                
+                return (TaskRunning(), VelocityCommand(velocity)) 
+            else:
+                return (TaskAborted(msg='Illegal state reached in Velocity test task'),)
+
+    def cancel(self):
```

Comment:
The only way for this task to exit is by it being canceled. I was hoping that the action message that starts this could set a time period to maintain the velocity for.